### PR TITLE
Prepare release 0.10.0

### PR DESCRIPTION
**Pull request by sjakobi**

### CODE REVIEW

**Comment by sjakobi** [comment]
While I strongly suspect that the pretty-printing functionality of the executable was affected by https://github.com/haskell-nix/hnix/pull/686#issuecomment-669101381, the library seems unchanged, so I went with a minor version bump.

**Comment by sjakobi** [comment]
There's an issue with some `aeson` instances, e.g.

```
src/Nix/Expr/Types.hs:546:10: error:
    • Overlapping instances for ToJSON NExpr
        arising from a use of ‘aeson-1.5.3.0:Data.Aeson.Types.ToJSON.$dmtoJSON’
      Matching instances:
        instance ToJSON1 f => ToJSON (Fix f)
          -- Defined in ‘aeson-1.5.3.0:Data.Aeson.Types.ToJSON’
        instance ToJSON NExpr -- Defined at src/Nix/Expr/Types.hs:546:10
    • In the expression:
        aeson-1.5.3.0:Data.Aeson.Types.ToJSON.$dmtoJSON @(NExpr)
      In an equation for ‘toJSON’:
          toJSON = aeson-1.5.3.0:Data.Aeson.Types.ToJSON.$dmtoJSON @(NExpr)
      In the instance declaration for ‘ToJSON NExpr’
    |
546 | instance ToJSON NExpr
    |          ^^^^^^^^^^^^
```

I'll try to fix that first.

File: CHANGELOG.md
Comment (Anton-Latukha) [inline CHANGELOG.md]: Please follow the already established structure.

To be understandable and explicit - state that this is a non-breaking change. It may be reasoned from the minor version number, but this style breaks the consistency of the changelog structure. During eading through the person observes structure inconsistency.

File: CHANGELOG.md
Comment (Anton-Latukha) [inline CHANGELOG.md]: There is tautological use of the word "dependency", section already names "Dependencies", so no reason to mention the topic of the section in the entries.

Comment (Anton-Latukha) [inline CHANGELOG.md]: The phrase "Update `package` dependency to `>= 1.7.0 && < 1.8`" is not sematically correct, since bounds are not a dependency.

What is updated is supported dependency versions, bounds of dependencies.

**Review by Anton-Latukha — CHANGES_REQUESTED** [review]
This is a test run of rigorous maintainer peer-review, the authoritative doc edition model where maintainer makes "suggestions" that block the PR and require the contributor to weasel the real PR improvements through the word policing process.

It was discussed in https://github.com/haskell-nix/hnix/issues/675#issuecomment-671833483 to employ this paradigm.

This is my real thought processes which I reserve and not force on others. But since closed doc edits paradigm requires to explicitly force them before merge, this is what shown here.

**Comment by Anton-Latukha** [comment]
Isn't that the above review process seems ridiculous and counterproductive?

Maintainer chooses the most counter-productive approach:

1. To just review and nitpick the text, and so force the person to comply own words to the maintainer liking the way maintainer particularly wants it. Essentially placing central scientific problems of semantics, social psychology and telecommunications - right in the middle of the process.

I've could:

2. Submit my edits into the PR.
3. Submit them after the merge.
4. Or make a correction Hackage release.

But maintainer decides to not use all 4 tools for the process flow, to use only the one, the most unproductive one.

**Review by Anton-Latukha — APPROVED** [review]

**Comment by Anton-Latukha** [comment]
We can move now with the release.

**Comment by Anton-Latukha** [comment]
`default.nix` dev env is not able to build inside `nixos-20.03` channel, but that is of no importance to the release process.
In reality `nixos-20.03` store currently does not hold any `0.9.1`, it holds `hnix 0.6.1`.

File: CHANGELOG.md
Comment (sjakobi) [inline CHANGELOG.md]: @Anton-Latukha, what does "Additional" mean here? Is this the heading for a section of _additions_ as opposed to _breaking changes_?

In this case, I think it is easy to misinterpret this wording as "additional dependencies".

How about something like this instead!?


```suggestion
* Dependency updates:
```

Comment (Anton-Latukha) [reply CHANGELOG.md]: Yep, `Additional`, since:
1. Only particular changes go into API breakages, and so in `Breaking` changes.
2. In "Minor" changes people include changes that are not breaking, it is one size fits all section, people are expected to include not only particular ones, but changelog expects to intake different things into it. `Minor` bears hidden knowledge of what `minor` means in Haskell, but Haskells `minor` does not include project-related changes and different stuff. `Different` - includes the breadth, but does not include specifics of application regarding the API, word does not underline that changes may be skipped over by the reader. So that is why `Additional`, word that means one-size-fits-all (`different`) & `minor` at the same time

Comment (Anton-Latukha) [reply CHANGELOG.md]: The changelog structure should be consistent, at least to respect the eye and the mind of the reader. So there always should be the same headlines that give the same most important message to the reader, an explicit `Breaking` section that includes the most important to reader information, and `Additional` that includes all other info project community conveys.

Comment (sjakobi) [reply CHANGELOG.md]: @Anton-Latukha The changelog structure that you seem to be describing sounds rather unconventional to me, so I think people might have difficulties understanding it.

In particular, using "additional" as a short-hand for "different" (I think this section is often labeled "other changes" in other changelogs) and "minor [changes]", seems extremely confusing to me.

Regarding consistency, I agree that it is important, but not more important than clarity.

Comment (Anton-Latukha) [reply CHANGELOG.md]: Yep, "other changes".

**Comment by sjakobi** [comment]
@Anton-Latukha I've tried to remove the points of contention from the changelog. How do you feel about this?

I'd really rather not get into another prolonged discussion about the changelog format though. If you want to make changes that I find hard to agree to, I'd rather close this PR and let you take care of the entire release.

**Comment by Anton-Latukha** [comment]
Why remove the points of contention. There was no contention before removals, I literally agreed with you here: https://github.com/haskell-nix/hnix/pull/689#discussion_r486380176:
> Yep, "other changes".

That responce concludded all topic we had here.

And before that I literally agreed here: https://github.com/haskell-nix/hnix/pull/689#issuecomment-674803184
> We can move now with the release.

That said that I agree with your process and I am out of the way so you can move with the release.

---

Now after the last change there is indeed contention - the structure is a one-level list of everything and their uncle, while it is a major release. Breaking changes not even enlisted first, how can this be a better form?

Thank you for this update of the content. But the catamorphism of the structure here...

Well.

Probably we misunderstood one another. Since I've made previous releases I waited for you to make this release. We communicated, you went with a changelog. I've wrote in the chatroom that I would be absent for some weeks.
 
Maybe you waited for me to merge this, maybe your busyness, IDK. The month went by.

Ok. Despite I see no point in cloning this PR, I would do as you said.

Thank you for updating it.

**Comment by Anton-Latukha** [comment]
Thank you.

Merged our changes co-authored.

Would make release right here.

**Comment by sjakobi** [comment]
> Why remove the points of contention. There was no contention before removals, I literally agreed with you here: [#689 (comment)](https://github.com/haskell-nix/hnix/pull/689#discussion_r486380176):
> 
> > Yep, "other changes".

I thought your `Yep, "other changes"` comment was just a confirmation that my assumption that, what you called "different" is the same as what other changelogs call "other changes", was correct.

If you wanted to agree to some change to the `Additional:` heading, I don't see how I ought to have understood that from that comment.

In general, I feel that your responses are often hard to understand in part because you usually don't include a reference to the particular bit that you're responding to.

I hope that at this level of miscommunication, you understand why I've become so reluctant to enter discussions with you.

**Comment by Anton-Latukha** [comment]
Thank you.

Great insight.

> In general, I feel that your responses are often hard to understand in part because you usually don't include a reference to the particular bit that you're responding to.

Ok. Noted.

**Pull request closed without merging**
